This notebook will clean the dataset where necessary, parse, and validate that the created columns are populating as expected.

In [46]:
import pandas as pd 
df = pd.read_csv("df_with_categorization.csv")

In [47]:
df_updated = df.drop(columns=["q1_away","q2_away","q3_away","q4_away","q1_home","q2_home","q3_home","q4_home","ot_away", "ot_home","h2_spread","h2_total","id_spread","id_total"])
df_updated.dtypes

season                      int64
date                       object
regular                      bool
playoffs                     bool
away                       object
home                       object
score_away                  int64
score_home                  int64
whos_favored               object
spread                    float64
total                     float64
moneyline_away            float64
moneyline_home            float64
underdog_wins                bool
underdog_covers_no_win     object
fave_covers                  bool
ATS_outcome                object
Season_Type                object
dtype: object

In [48]:
df_updated[["underdog_wins"]].value_counts()

underdog_wins
False            15803
True              7312
Name: count, dtype: int64

In [49]:
df_updated[["underdog_covers_no_win"]].value_counts()

underdog_covers_no_win
False                     18708
True                       4072
Push                        335
Name: count, dtype: int64

In [50]:
df_updated[["fave_covers"]].value_counts()

fave_covers
False          11719
True           11396
Name: count, dtype: int64

In [51]:
df_updated.isna().sum()

season                       0
date                         0
regular                      0
playoffs                     0
away                         0
home                         0
score_away                   0
score_home                   0
whos_favored                 0
spread                       0
total                        0
moneyline_away            3298
moneyline_home            3298
underdog_wins                0
underdog_covers_no_win       0
fave_covers                  0
ATS_outcome                  0
Season_Type                  0
dtype: int64

Checking that all values have been categorized:<br>
<u>underdog_wins</u><br>
15803(False) + 7312(True) = 23115<br>
<u>underdog_covers_no_win</u><br>
18708(False) + 4072 (True) + 335 (Push) = 23115<br>
<u>fave_covers</u><br>
11719(False) + 11396(True) = 23115<br>
7312 (underdog wins) + 4072 (underdog cover, no win) + 335 (Push) + 11396 (fave covers) = 23115


In [52]:
mask_regular= df_updated["regular"] == True
mask_playoffs = df_updated["playoffs"] == True

regular_season_df = df_updated[mask_regular]
playoff_df = df_updated[mask_playoffs]

regular_season_df.to_csv("RS_df.csv", index = False)
playoff_df.to_csv("playoff_df.csv", index = False)


In [53]:
df_updated["playoffs"].value_counts()

playoffs
False    21608
True      1507
Name: count, dtype: int64

In [54]:
df_updated["regular"].value_counts()

regular
True     21575
False     1540
Name: count, dtype: int64

In [55]:
result = df_updated[(df_updated["playoffs"] == False) & (df_updated["regular"] == False)]
result.head(10)

,season,date,regular,playoffs,away,home,score_away,score_home,whos_favored,spread,total,moneyline_away,moneyline_home,underdog_wins,underdog_covers_no_win,fave_covers,ATS_outcome,Season_Type
16579,2020,2020-08-15,False,False,mem,por,122,126,home,6.0,229.5,220.0,-260.0,False,True,False,Underdog Covers But Loses,PlayIn
17743,2021,2021-05-18,False,False,cha,ind,117,144,home,1.5,226.0,110.0,-130.0,False,False,True,Favorite Covers,PlayIn
17744,2021,2021-05-18,False,False,wsh,bos,100,118,home,2.0,233.0,110.0,-130.0,False,False,True,Favorite Covers,PlayIn
17745,2021,2021-05-19,False,False,sa,mem,96,100,home,4.5,224.0,170.0,-190.0,False,True,False,Underdog Covers But Loses,PlayIn
17746,2021,2021-05-19,False,False,gs,lal,100,103,home,5.5,217.0,205.0,-245.0,False,True,False,Underdog Covers But Loses,PlayIn
17747,2021,2021-05-20,False,False,ind,wsh,115,142,home,3.5,240.5,140.0,-160.0,False,False,True,Favorite Covers,PlayIn
17748,2021,2021-05-21,False,False,mem,gs,117,112,home,4.5,220.5,185.0,-215.0,True,False,False,Underdog Wins,PlayIn
19062,2022,2022-04-12,False,False,cle,bkn,108,115,home,9.5,227.5,400.0,-500.0,False,True,False,Underdog Covers But Loses,PlayIn
19063,2022,2022-04-12,False,False,lac,min,104,109,home,3.5,231.0,135.0,-155.0,False,False,True,Favorite Covers,PlayIn
19064,2022,2022-04-13,False,False,cha,atl,103,132,home,5.5,235.0,185.0,-215.0,False,False,True,Favorite Covers,PlayIn


In [56]:
regular_season_df.shape

(21575, 18)

In [57]:
playoff_df.shape

(1507, 18)

There are 33 games unaccounted for. These games were apart of the play-in tournament, starting the 2020 season. We will make a separate dataframe to represent these, as well as a dataframe that will combine playoffs and play-in tournament.

In [58]:
play_in_tournament_df = result
play_in_tournament_df.to_csv("play_in.csv",index = False)
play_off_in_combined_mask = (df_updated["regular"] == False) & (df_updated["playoffs"] == False)
playoff_in_df = df[play_off_in_combined_mask]
playoff_in_df.to_csv("playoff_in_df.csv",index = False)

In [59]:
regular_season_df["spread"].value_counts()

spread
2.5     1315
3.5     1310
4.5     1298
5.5     1213
6.5     1163
1.5     1110
7.5     1012
3.0     1003
2.0      991
4.0      978
5.0      917
6.0      885
7.0      841
8.5      814
8.0      694
9.5      693
1.0      649
9.0      619
10.5     534
10.0     494
11.5     450
11.0     369
12.5     351
13.5     280
12.0     268
0.0      239
13.0     196
14.5     169
14.0     164
15.5     120
15.0      91
16.5      80
0.5       79
16.0      45
17.0      38
17.5      32
18.0      22
18.5      17
19.5      14
19.0       6
21.0       3
21.5       3
20.0       2
20.5       2
23.0       1
23.5       1
Name: count, dtype: int64